config.py

```diff
@@ -0,0 +1,102 @@
+from configparser import ConfigParser
+
+class GlobalOptions():
+    ''' Container class for global options
+    '''
+    def __init__(self, port, mcast_link, mcast_site, default_domain, default_domain_type, ipv4_gateway):
+        self.port = port
+        self.mcast_link = mcast_link
+        self.mcast_site = mcast_site
+        self.default_domain = default_domain
+        self.default_domain_type = default_domain_type
+        self.ipv4_gateway = ipv4_gateway
+
+class DomainOptions():
+    ''' Base container class for per domain options
+    '''
+    @classmethod
+    def from_parser(cls, section, parser, globals):
+        ''' Builds a DomainOptions object from a config section
+            Handles domain type specific options automatically
+        '''
+        from domain import DomainType
+
+        type = parser.get(section, 'DomainType', fallback=globals.default_domain_type)
```

Comment:
shadowing the type() function is bad practice :)

Reply:
True, fixed